File: RateMyP/RateMyP.WebApp/ClientApp/src/extensions/StringExtensions.ts
export {};

var RegexMatches: [string, RegExp][] = [
    [ 'a', /[ą]/g ],
    [ 'A', /[Ą]/g ],
    [ 'c', /[č]/g ],
    [ 'C', /[Č]/g ],
    [ 'e', /[ęė]/g ],
    [ 'E', /[ĘĖ]/g ],
    [ 'i', /[į]/g ],
    [ 'I', /[Į]/g ],
    [ 's', /[š]/g ],
    [ 'S', /[Š]/g ],
    [ 'u', /[ųū]/g ],
    [ 'U', /[ŲŪ]/g ],
    [ 'z', /[ž]/g ],
    [ 'Z', /[Ž]/g ],
];

declare global {
    interface String {
        denationalize(): string;
    }
}

String.prototype.denationalize = function(): string {
    var str = this;
    RegexMatches.forEach(pair => {
        var letter = pair["0"];
        var regex = pair["1"];
        str = str.replace(regex, letter);
    })
    return str as unknown as string
}

File: WinForms-deprecated/RateMyP.Client/Helpers/StringExtensions.cs
using System.Collections.Generic;
using System.Text.RegularExpressions;

namespace RateMyP.Helpers
    {
    public static class StringExtensions
        {
        private static Dictionary<char, string> s_regexes = new Dictionary<char, string>
            {
                { 'a', "[ą]" },
                { 'A', "[Ą]" },
                { 'c', "[č]" },
                { 'C', "[Č]" },
                { 'e', "[ęė]" },
                { 'E', "[ĘĖ]" },
                { 'i', "[į]" },
                { 'I', "[Į]" },
                { 's', "[š]" },
                { 'S', "[Š]" },
                { 'u', "[ųū]" },
                { 'U', "[ŲŪ]" },
                { 'z', "[ž]" },
                { 'Z', "[Ž]" },
            };

        public static string Denationalize(this string str)
            {
            foreach (var pair in s_regexes)
                {
                var letter = pair.Key.ToString();
                var regex = pair.Value;
                str = Regex.Replace(str, regex, letter);
                }
            return str;
            }
        }
    }
